src/batInnerWeaktbl.ml

```diff
@@ -95,6 +95,7 @@ module Make (H: Hashtbl.HashedType) : Hashtbl.S with type key = H.t = struct
   let find_all tbl key =
     try all_value (W.find tbl (dummy key)) with Not_found-> []
   let find tbl key = top_value (W.find tbl (dummy key))
+  let find_opt tbl key = try Some (top_value (W.find tbl (dummy key))) with Not_found -> None
```

Comment:
maybe you could write `try (Some (find tbl key)) ...` here? The code is a bit complex so I'd rather stay simple.